### DISCUSSION TITLE
SAML pull request - how to test

### DISCUSSION THREAD

**Discussion by RadekVoltr** [simpleidserver/SimpleIdServer#250] [General]
Hi

I am working on set of changes to allow SimpleIdServer SAML IdP pass through https://samltest.id/ test tool.

I have done changes so code is now able to work with SP using Http-Post and not just Http-Redirect. But I need to verify Http-Redirect before I create PR as I did couple of changes in code.

Acceptance Tests in solution passing. Can you tell me how I should verify SP with Http-Redirect ? Do you have some tip for online testing server like SamlTest.Id with Http-Redirect usage ?

Thanks Radek

**Comment by simpleidserver** [comment]
Hello,

First, thank you a for your contribution :)
It's possible to simulate HTTP-REDIRECT instruction by using keycloak. 
Follow the steps below :

1. Install keycloak with docker : 

`docker run -p 8080:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:17.0.1 start-dev`

2. Navigate to the website : "http://localhost:8080".
3. Authenticate with the credential : admin, admin
4. Click on "Identity Provider" and add a new "SAML2.0" identity provider. In the edition screen you can choose between HTTP-POST & HTTP-GET binding :).
5. Update the Single Sign-On Service URL : http://localhost:7000/saml/SSO/Login.
6. Upload the metadata file.
7. Run your SAML IDP.
8. Disconnect from the portal and click on "saml".

**Comment by RadekVoltr** [comment]
Tested & PR created. Fixed also couple of other minor issues which created issues with last KeyCloak version on http-redirect

**Reply by simpleidserver** [reply]
Hello thank you a lot for your pull request, i'm going to review it.

**Reply by simpleidserver** [reply]
I reviewed and made some modifications on your changes :)
The SAMLIDP project (branch release/2.0.6) should work with KeyCloak & samltest.id.

Kind Regards,

SimpleIdServer

**Reply by RadekVoltr** [reply]
Well, I expected that :-D

best regards
Radek
